nexus/catalog/src/lib.rs

```diff
@@ -602,3 +594,80 @@ impl Catalog {
         })
     }
 }
+
+// This is mostly copied from peer-postgres
+#[async_trait::async_trait]
+impl QueryExecutor for Catalog {
+    #[tracing::instrument(skip(self, stmt), fields(stmt = %stmt))]
+    async fn execute(&self, stmt: &Statement) -> PgWireResult<QueryOutput> {
```

Comment:
is there a way to refactor to share this logic between postgres peer impl and this?